utils/core/src/serde/byte_writer.rs

```diff
@@ -83,3 +103,22 @@ impl ByteWriter for Vec<u8> {
         self.extend_from_slice(values);
     }
 }
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Returns the length of the value in vint64 enoding.
+pub fn encoded_len(value: usize) -> usize {
+    match value.leading_zeros() {
+        0..=7 => 9,
+        8..=14 => 8,
+        15..=21 => 7,
+        22..=28 => 6,
+        29..=35 => 5,
+        36..=42 => 4,
+        43..=49 => 3,
+        50..=56 => 2,
+        57..=64 => 1,
+        _ => unreachable!(),
+    }
```

Comment:
Would this be equivalent to something like this:
```Rust
let zeros = value.leading_zeros() as usize;
let len = zeros.saturating_sub(1) / 7;
9 - core::cmp::min(len, 8)
```

Reply:
Indeed! I suppose, that this implementation works a little bit slower, but it looks much better for sure!

Reply:
I didn't benchmark, but looking at the generated assembly, I think this should actually be faster as compared to the `match` approach. Or, at least, there is much less branching in the logic.